packages/exchange/src/exchange/providers/openai.py

```diff
@@ -39,11 +40,11 @@ def __init__(self, client: httpx.Client) -> None:
     @classmethod
     def from_env(cls: Type["OpenAiProvider"]) -> "OpenAiProvider":
         cls.check_env_vars(cls.instructions_url)
-        url = os.environ.get("OPENAI_HOST", OPENAI_HOST)
+        url = get_env_url("OPENAI_HOST", OPENAI_HOST)
         key = os.environ.get("OPENAI_API_KEY")
 
         client = httpx.Client(
-            base_url=url + "v1/",
+            base_url=url.join("v1/"),
```

Comment:
fyi using httpx.URL as it isn't so sensitive about trailing slash etc when joining

Reply:
interesting, i always remembered `//` would resolve? just curious what bug you can into since i haven't seen this before!

Reply:
I think it was more about not having a slash at all. e.g if you set a base URL ending in the port